### ISSUE TITLE
bsdthread_register error after OS update to macOS Monterey (12.2.1)

### ISSUE THREAD

**Issue by totoroot** [openshift/oc#1079]
**Describe the bug**
Launching oc causes bsdthread_register error.

This issue only came up after an operating system update to macOS 12.2.1 (21D62), Monterey. It was running fine under macOS Big Sur until workplace device policies forced me to update the OS.

**To Reproduce**
1. Launch oc from kitty terminal emulator or iTerm2 on macOS Monterey
2. See error

**Error**
```
fatal error: runtime: bsdthread_register error

runtime stack:
runtime.throw(0x37ead3b, 0x21)
	/usr/local/go/src/runtime/panic.go:616 +0x81 fp=0x7ff7bfefef38 sp=0x7ff7bfefef18 pc=0x102ae91
runtime.goenvs()
	/usr/local/go/src/runtime/os_darwin.go:129 +0x83 fp=0x7ff7bfefef68 sp=0x7ff7bfefef38 pc=0x1028a13
runtime.schedinit()
	/usr/local/go/src/runtime/proc.go:501 +0xd6 fp=0x7ff7bfefefd0 sp=0x7ff7bfefef68 pc=0x102d786
runtime.rt0_go(0x7ff7bfeff008, 0x3, 0x7ff7bfeff008, 0x0, 0x1000000, 0x3, 0x7ff7bfeff270, 0x7ff7bfeff273, 0x7ff7bfeff27e, 0x0, ...)
	/usr/local/go/src/runtime/asm_amd64.s:252 +0x1f4 fp=0x7ff7bfefefd8 sp=0x7ff7bfefefd0 pc=0x1056af4
```

**Environment details**
```
Darwin 21.3.0 Darwin Kernel Version 21.3.0: Wed Jan  5 21:37:58 PST 2022; root:xnu-8019.80.24~20/RELEASE_X86_64 x86_64
ProductName:	macOS ProductVersion:	12.2.1 BuildVersion:	21D62
```

**Comment by totoroot** [comment]
Since I cannot run `oc version` without getting the fatal erroe, I cannot say which version I've got installed.
I've updated it to the latest version with brew, so according to [https://formulae.brew.sh/formula/openshift-cli#default](brew.sh) it should be 4.6.0

**Comment by guli-bn** [comment]
Hi @totoroot, I'm here for just the same reason. I just now managed to build my own working version with a correct version of go:

`brew install go` (for latest go, that's whats making it crash)

And then checkout this repo and follow building instructions in the readme: https://github.com/openshift/oc#building

The binary gets placed in the root of the repository and needs to be copied to your path (you might have to uninstall brew-oc)

**Comment by totoroot** [comment]
@guli-bn Thanks, building from source with go version 1.17.8 worked for me too.
I'm leaving this issue open until homebrew's version gets fixed.

**Comment by totoroot** [comment]
This seems to be fixed in openshift-cli version 4.6.0 installed via homebrew.

**Comment by manuartero** [comment]
> This seems to be fixed in openshift-cli version 4.6.0 installed via homebrew.

This worked for me, thanks! 

```bash
brew reinstall openshift-cli
```
